lib/controller.js

```diff
@@ -176,6 +177,62 @@ may or may not be open and closes them
 
 */
 
+/*
+Reboots your tessel
+*/
+
+controller.reboot = function(opts) {
+  return new Promise(function(resolve, reject) {
+    Tessel.get(opts)
+      .then(function(tessel) {
+        var command = 'ssh root@' + tessel.lanConnection.ip + ' -i ' + opts.key + ' \'reboot ';
+        if (opts.time ||  opts.delay)  {
+          command = command + ' -d ' + opts.time || opts.delay;
+        }
+        if (opts.force) {
+          command = command + ' -f ';
+        }
+        if (opts.nosync) {
+          command = command + ' -n ';
+        }
+        command = command + '\'';
+        debug(command);
+        var reboot_cp = cp.exec(command);
+
+        reboot_cp.on('error', function(e) {
+          if (e === 255) {
+            logs.warn('Not authorized (maybe wrong key)');
+            reject();
+          } else if (e === 127) {
+            // command not found
+            logs.warn('Command "reboot" not found!');
+            reject();
+          } else {
+            reject(e);
+          }
+        });
+        reboot_cp.on('exit', function(e) {
+          if (e === 255) {
+            logs.warn('Not authorized (maybe wrong key)');
+            reject();
+          } else if (e === 127) {
+            // command not found
+            logs.warn('Command "reboot" not found!');
+            reject();
+          } else if (e === 0) {
+            logs.info(tessel.name + ' is going to reboot!');
+            resolve();
+          } else {
+            reject(e);
+          }
```

Comment:
Could these be consolidated into a single handler that gets share by both events?